includes/class-plugin.php

```diff
@@ -571,6 +579,66 @@ public function get_redis_status() {
         return $wp_object_cache->redis_status();
     }
 
+    /**
+     * Build the connection parameters from config constants.
+     *
+     * @return array
+     */
+    public function build_parameters() {
+        $parameters = [
+            'scheme' => 'tcp',
+            'host' => '127.0.0.1',
+            'port' => 6379,
+            'database' => 0,
+            'timeout' => 1,
+            'read_timeout' => 1,
+            'retry_interval' => null,
+            'persistent' => false,
+        ];
+
+        $settings = [
+            'scheme',
```

Comment:
The only feedback I have is this function seems unnecessary on the `Plugin`, I'll move it to `Predis.`